spec/services/invoices/payments/adyen_service_spec.rb

```diff
@@ -0,0 +1,244 @@
+        expect(result.invoice.payment_attempts).to eq(1)
+        expect(result.invoice.reload.ready_for_payment_processing).to eq(false)
+
+        expect(result.payment.id).to be_present
+        expect(result.payment.invoice).to eq(invoice)
+        expect(result.payment.payment_provider).to eq(adyen_payment_provider)
+        expect(result.payment.payment_provider_customer).to eq(adyen_customer)
+        expect(result.payment.amount_cents).to eq(invoice.total_amount_cents)
+        expect(result.payment.amount_currency).to eq(invoice.currency)
+        expect(result.payment.status).to eq('Authorised')
+
+        expect(adyen_customer.reload.payment_method_id).
+          to eq(payment_methods_response.response['storedPaymentMethods'].first['id'])
+      end
+
+      expect(payments_api).to have_received(:payments)
+    end
+
+    context 'with no payment provider' do
+      let(:adyen_payment_provider) { nil }
+
+      it 'does not creates a adyen payment' do
+        result = adyen_service.create
+
+        expect(result).to be_success
+
+        aggregate_failures do
+          expect(result.invoice).to eq(invoice)
+          expect(result.payment).to be_nil
+
+          expect(payments_api).not_to have_received(:payments)
+        end
+      end
+    end
+
+    context 'with 0 amount' do
+      let(:invoice) do
+        create(
+          :invoice,
+          organization:,
+          customer:,
+          total_amount_cents: 0,
+          currency: 'EUR',
+        )
+      end
+
+      it 'does not creates a adyen payment' do
+        result = adyen_service.create
+
+        expect(result).to be_success
+
+        aggregate_failures do
+          expect(result.invoice).to eq(invoice)
+          expect(result.payment).to be_nil
+
+          expect(result.invoice).to be_succeeded
+
+          expect(payments_api).not_to have_received(:payments)
+        end
+      end
+    end
+
+    context 'when customer does not have a provider customer id' do
+      before { adyen_customer.update!(provider_customer_id: nil) }
+
+      it 'does not creates a adyen payment' do
+        result = adyen_service.create
+
+        expect(result).to be_success
+
+        aggregate_failures do
+          expect(result.invoice).to eq(invoice)
+          expect(result.payment).to be_nil
+
+          expect(payments_api).not_to have_received(:payments)
+        end
+      end
+    end
+
+    context 'with error on adyen' do
+      let(:customer) { create(:customer, organization:) }
+
+      let(:subscription) do
+        create(:subscription, organization:, customer:)
+      end
+
+      let(:organization) do
+        create(:organization, webhook_url: 'https://webhook.com')
+      end
+
+      before do
+        subscription
+
+        allow(payments_api).to receive(:payments)
+          .and_raise(Adyen::AdyenError.new(nil, nil, 'error', 'code'))
+      end
+
+      it 'delivers an error webhook' do
+        expect { adyen_service.send(:create_adyen_payment) }
```

Comment:
Style/Send: Prefer `Object#__send__` or `Object#public_send` to `send`. (https://rubystyle.guide#prefer-public-send)